### PR TITLE
Fix/double bookings

### PR DESCRIPTION
**Pull request by horpeazy**
## What does this PR do?

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Tests (Unit/Integration/E2E or any other test)
- This change requires a documentation update

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/app-store/zohocalendar/lib/CalendarService.ts [hunk 1]
```diff
@@ -266,8 +267,8 @@ export default class ZohoCalendarService implements Calendar {
         .filter((freebusy: FreeBusy) => freebusy.fbtype === "busy")
         .map((freebusy: FreeBusy) => ({
           // using dayjs utc plugin because by default, dayjs parses and displays in local time, which causes a mismatch
-          start: dayjs.utc(freebusy.startTime, "YYYYMMDD[T]HHmmss[Z]").toISOString(),
-          end: dayjs.utc(freebusy.endTime, "YYYYMMDD[T]HHmmss[Z]").toISOString(),
+          start: moment.utc(freebusy.startTime, "YYYYMMDDTHHmmssZ").toISOString(),
+          end: moment.utc(freebusy.endTime, "YYYYMMDDTHHmmssZ").toISOString(),
```
Comment (GAMHnile) [inline packages/app-store/zohocalendar/lib/CalendarService.ts:271]: @horpeazy why did you change to using moment

Comment (horpeazy) [reply packages/app-store/zohocalendar/lib/CalendarService.ts:271]: @GAMHnile dayJS did not work, I kept getting this error "RangeError: Invalid time value", the reason was because it could not parse the date with the given format. I made an adjustment to make this work, but ran into another issue where ZOHO APIs was not returning a consistent format, for some time slots it did not not include time(just date) and this caused another issue. "moment" was able to handle it, so I just used "moment" to keep things simple and not account for difference scenarios "moment" probably has covered already.


Examples:
20241001T160000Z(With time)
20241005Z(Without time)
